zone/pets.cpp

```diff
@@ -164,6 +165,17 @@ void Mob::MakePoweredPet(uint16 spell_id, const char* pettype, int16 petpower,
 	// 4 - Keep DB name
 	// 5 - `s ward
 
+	const std::vector<CharacterPetNameRepository::CharacterPetName>& vanity_name = CharacterPetNameRepository::GetWhere(
```

Comment:
Since `char_id` is the primary key we can use `FindOne` to return a single entry and then just check if `e.char_id != 0` or whatever.